### PR TITLE
Oracle work in progress

### PR DESCRIPTION
**Pull request by JasperSlimmens**
I will work on Oracle in this branch.

### CODE REVIEW

**Comment by JasperSlimmens** [comment]
LPcheck functions working. For now I have worked around checking UniLp, because it takes really long to run. Will look into that later.

getUnderlying functions for all LPs working. Had to work some tricks for Curve as their Registry has quite some mistakes.

File: contracts/curve/interfaces/IRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
+pragma solidity >=0.5.0;
+
+interface IRegistry {
+  function get_n_coins(address _pool) external view returns (uint256[2] memory);
```
Comment (belbix) [inline contracts/curve/interfaces/IRegistry.sol:4]: please, follow official style guide and use mixedCase
https://docs.soliditylang.org/en/v0.5.3/style-guide.html#function-names

Comment (JasperSlimmens) [reply contracts/curve/interfaces/IRegistry.sol:4]: This is interface with Curve registry. https://etherscan.io/address/0x7d86446ddb609ed0f5f8684acf30380a356b2b4c#readContract

They name their functions with underscores

Comment (JasperSlimmens) [reply contracts/curve/interfaces/IRegistry.sol:4]: I can rename the interface to ICurveRegistry if that is more clear

Comment (belbix) [reply contracts/curve/interfaces/IRegistry.sol:4]: ahah, yeah, better rename

**Comment by JasperSlimmens** [comment]
Oracle contract is now fully functional. I am running many test to find and fix bugs. Getting prices for LP tokens works well. Some tokens seem to give errors due to bad contracts on their end. These are not really a problem as they are unknown coins without value. Will work on a proper test to show functionality and also test accuracy, comparing with Coingecko.

Control functions still need to be added.

**Comment by JasperSlimmens** [comment]
Test results:

https://cdn.discordapp.com/attachments/807929064220131329/811889732925390888/unknown.png
https://cdn.discordapp.com/attachments/807929064220131329/811889881970245662/unknown.png
https://cdn.discordapp.com/attachments/807929064220131329/811890006666379274/unknown.png
https://cdn.discordapp.com/attachments/807929064220131329/811890110505156608/unknown.png
https://cdn.discordapp.com/attachments/807929064220131329/811890222396211200/unknown.png
https://cdn.discordapp.com/attachments/807929064220131329/811890349197885460/unknown.png

File: contracts/Oracle.sol [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+
+  //Checks if address is 1Inch LP
+  function isOneInchCheck(address token) public view returns (bool) {
+    bool oneInchLP = oneInchFactory.isPool(token);
+    return oneInchLP;
+  }
+
+  //Checks if address is Uni LP. This is done in two steps, because the second step seems to cause errors for some tokens.
+  //Only the first step is not deemed accurate enough, as any token could be caalled UNI-V2.
+  function isUniCheck(address token) public view returns (bool) {
```
Comment (belbix) [inline contracts/Oracle.sol:151]: looks like we can unite it with isSushiCheck, just have the symbol as an attribute

**Pull request merged**